src/table/context/TableContext.tsx

```diff
@@ -51,6 +52,10 @@ export const TableContextProvider = ({
     tableWidth,
     styling
   );
+  const showRightBorder = useMemo(
```

Comment:
This a case that is not considered here and that is when the use have a vertical scrollbar that take up space.

So you probably want to move this into `useColumnWidth` so that you can include the `yScrollbarWidth`. Otherwise it will show a right border for the last column even if the columns occupies all width available.